functions/extend/ImagifyExtension.php

```diff
@@ -0,0 +1,101 @@
+<?php
+namespace crop_thumbnails\extend;
+
+/**
+ * Imagify is added for compatibility reasons.
+ * This class will trigger the Imagify optimization when a new thumbnail is created. 
+ * Plugin-URL: https://en-gb.wordpress.org/plugins/imagify/
+ * 
+ * @author Aaron Summers
+ * @link https://github.com/aaronsummers
+ * @version 1.0.0
+ * @since 1.8.0
+ * @package crop-thumbnails
+ */
+class ImagifyExtension
+{
+	 public static function doExtend() {
+
+		// crop_thumbnails_after_save_new_thumb is a hook that is triggered after a new thumbnail is created amd returns the file path of the new thumbnail (not the URL)
```

Comment:
@aaronsummers thank you for your code.

Can you please make some small adjustments?
The add_action is always triggered no matter what the Imagify_Auto_Optimization exists.

I would love to see the something linke this:
```
add_action('plugins_loaded', function() {
  if( class_exists( Imagify_Auto_Optimization::class) ) {
    add_action( 'crop_thumbnails_after_save_new_thumb', array( self::class, 'actionDoWebPConvertion' ), 100, 1 );
  }
});
```

I know you checked in actionDoWebPConvertion, but this way not even the action-hook is triggered if the plugin do not exists.

Reply:
This has been updated.

Reply:
Not sure if you need me to mark this as resolved @vollyimnetz ?

Reply:
I am on it.